wd_helper.au3

```diff
@@ -538,48 +538,78 @@ EndFunc   ;==>_WD_IsWindowTop
 ; Description ...: Enter the specified frame.
 ; Syntax ........: _WD_FrameEnter($sSession, $vIdentifier)
 ; Parameters ....: $sSession    - Session ID from _WD_CreateSession
-;                  $vIdentifier - Index (as 0-based Integer) or Element ID (as String) or Null (Keyword)
+;                  $vIdentifier - Target frame identifier. Can be any of the following:
+;                  |Null    - Return to top-most browsing context
+;                  |String  - Element ID from _WD_FindElement or path like 'null/2/0'
+;                  |Integer - 0-based index of frames
 ; Return values .: Success - True.
-;                  Failure - WD Response error message (E.g. "no such frame") and sets @error to $_WD_ERROR_Exception
+;                  Failure - WD Response error message (E.g. "no such frame") and sets @error to $_WD_ERROR_Exception or $_WD_ERROR_InvalidArgue
 ; Author ........: Decibel
 ; Modified ......: mLipok
-; Remarks .......: You can drill-down into nested frames by calling this function repeatedly with the correct parameters.
+; Remarks .......: You can drill-down into nested frames by calling this function repeatedly with the correct parameters or use path like 'null/2/0'
 ; Related .......: _WD_Window, _WD_LastHTTPResult
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
 Func _WD_FrameEnter($sSession, $vIdentifier)
 	Local Const $sFuncName = "_WD_FrameEnter"
-	Local Const $sParameters = 'Parameters:    Identifier=' & $vIdentifier
-	Local $sOption, $sValue, $sResponse, $oJSON
+	If String($vIdentifier) = 'null' Then $vIdentifier = Null ; String must be used because checking 0 = 'null' is True
+	Local Const $bIsIdentifierNull = (IsKeyword($vIdentifier) = $KEYWORD_NULL)
+	Local Const $sParameters = 'Parameters:    Identifier=' & ($bIsIdentifierNull ? ("Null") : ($vIdentifier))
+	Local $sValue, $sMessage = '', $sOption, $sResponse, $oJSON
 	Local $iErr = $_WD_ERROR_Success
 
 	;*** Encapsulate the value if it's an integer, assuming that it's supposed to be an Index, not ID attrib value.
-	If (IsKeyword($vIdentifier) = $KEYWORD_NULL) Then
+	Local Const $aIdentifiers = StringSplit($vIdentifier, '/')
+	Local Const $bIdentifierAsPath = ($aIdentifiers[0] > 1)
+	If $bIdentifierAsPath Then
+		For $i = 1 To $aIdentifiers[0]
+			If String($aIdentifiers[$i]) = 'null' Then
+				$sValue = _WD_FrameEnter($sSession, Null)
+				$iErr = @error
+			ElseIf StringIsDigit($aIdentifiers[$i]) And IsInt(Number($aIdentifiers[$i])) Then
+				$sValue = _WD_FrameEnter($sSession, Int($aIdentifiers[$i]))
+				$iErr = @error
+			Else
+				$iErr = $_WD_ERROR_InvalidArgue
+			EndIf
+			If $iErr Then ExitLoop
+		Next
+		If $i > $aIdentifiers[0] Then $i = $aIdentifiers[0]
+		If $iErr Then $sMessage = ' Error on ID#' & $i
```

Comment:
```autoit
If $i > $aIdentifiers[0] Then $i = $aIdentifiers[0]
```
My intention was the case when the entire loop was procesed to set properly $sMessage

But after rethinking `If $i > $aIdentifiers[0]` will happen only when there will not be `$iErr` set.
 So there is no need to check it as 
```autoit
If $iErr Then $sMessage = ' Error on ID#' & $i
```
can happen but not in case `If $i > $aIdentifiers[0]`

Reply:
> So there is no need to check it

@Danp2  Do you agree ?

Reply:
> 		If $iErr Then $sMessage = ' Error on ID#' & $i

This could be moved inside your For...Next loop where you are already checking $iErr

Reply:
P.S. This is another PR that has run amok.

Reply:
From my perspective, we are simply creating a new functionality that requires multiple exchanges of views and experiences.

Reply:
Unfortunately, we have different approaches, experiences, and it takes time and patience to create added value together.

Reply:
Isn't that normal in a creative process involving different people from far away places, especially not in direct contact with each other?

Reply:
> > ```
> >   If $iErr Then $sMessage = ' Error on ID#' & $i
> > ```
> 
> This could be moved inside your For...Next loop where you are already checking $iErr

done with: [no more recursively calling _WD_FrameEnter](https://github.com/Danp2/au3WebDriver/pull/364/commits/c183e5c9955fc5d0c3a3401773a7611ea7dc9468)

Reply:
Perhaps. I get easily confused when there are multiple interdependent PRs and multiple separate conversations occurring simultaneously. FWIW, I generally don't open a PR until the coding is "review ready".

Reply:
> I get easily confused when there are multiple interdependent PRs and multiple separate conversations occurring simultaneously. 

I find it hard to get to that too sometimes.
But I have it so much easier that I set the direction of changes in a given PR, and you are just trying to understand me and direct me on the right track.
As a result, it is relatively easier for me to grasp this rush of changes, because I am their author.

> FWIW, I generally don't open a PR until the coding is "review ready".

Hmm .. I would like that too, but it is possible when you change something small, or introduce new functionalities "for your own".
When you create code for your requirements, you can create it right away.
When I create the code, I have to modify it and change my approach due to your expectations, which are announced to me only after I present my version to you.

I just can't guess what code you will finally accept.
I create it in the best shape I can think of.

When I publish a new PR. This is where the constructive exchange of ideas and concepts begins.

Hope you get used to it.

Reply:
I wouldn't count on it. 😦